components/button/tests/unit/button.spec.js

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import { Button } from '@westpac/button';
+import { GEL } from '@westpac/core';
+import { fireEvent, render, screen, within } from '@testing-library/react';
+import { overridesTest } from '../../../../helpers/tests/overrides-test.js';
+import { nestingTest } from '../../../../helpers/tests/nesting-test.js';
+import wbc from '@westpac/wbc';
+import { ArrowLeftIcon } from '@westpac/icon';
+
+// The default tests every component should run
+overridesTest({
+	name: 'button', // the name has to be the package name without '@westpac/' scope
+	overrides: ['Button'], // every single override root key
+	Component: (props) => <Button {...props}>Button content</Button>, // the component with all components rendered
+});
+
+// another default test to check that the component errors when outside of GEL and renders when inside
+nestingTest({
+	name: 'button',
+	Component: (props) => <Button {...props} />,
+});
+
+const SimpleButton = (props) => (
+	<GEL brand={wbc}>
+		<Button {...props}>{props.children}</Button>
+	</GEL>
+);
+
+// Component specific tests
+describe('Button component', () => {
+	const originalLog = console.log;
+	beforeEach(() => {
+		console.log = jest.fn();
+	});
+	afterEach(() => {
+		console.log = originalLog;
+	});
+
+	test('should be disabled when prop "disabled" is passed', () => {
+		const someText = 'Hello World';
+		const someContent = <span data-testid="my-span">{someText}</span>;
+		const { container, getByRole } = render(<SimpleButton disabled>{someContent}</SimpleButton>);
+		expect(getByRole('button')).toBeInTheDocument();
+		expect(getByRole('button')).toHaveAttribute('disabled');
+		const theButton = container.querySelector('button[disabled]');
+		expect(theButton).toBeInTheDocument();
+		expect(theButton).toHaveTextContent(someText);
+	});
+
+	test('should show normal, default, button color when not active', () => {
+		const { getByTestId } = render(<SimpleButton data-testid="my-button" />);
+		const theButton = getByTestId('my-button');
+		// Property 'look' has default value 'hero', so check for the hero color
+		expect(theButton).toHaveStyle('background-color: rgb(31, 28, 79)');
```

Comment:
Colour theme can change. Please don't use colours directly in unit tests. We should let the UX to change the colours when it's needed.